docs/writing-rules/testing-rules.md

```diff
@@ -69,82 +91,115 @@ eval(safe_get_user_input())
 Run the tests with the following:
 
 ```sh
-python -m semgrep --quiet --test rules/
+semgrep --test rules/
 ```
 
 Which will produce the following output:
 ```sh
-1 yaml files tested
-check id scoring:
---------------------------------------------------------------------------------
-(TODO: 2) rules/detect-eval.yaml
-	✖ insecure-eval-use                                  TP: 1 TN: 2 FP: 1 FN: 1
-	test: rules/detect-eval.py, expected lines: [5, 12], reported lines: [5, 15]
---------------------------------------------------------------------------------
-final confusion matrix: TP: 1 TN: 2 FP: 1 FN: 1
---------------------------------------------------------------------------------
+1/1: ✓ All tests passed
+No tests for fixes found.
 ```
 
-- True positives (`TP`) correspond to `ruleid`
-- True negatives (`TN`) correspond to `ok`
-- False positives (`FP`) correspond to `todook`
-- False negatives (`FN`) correspond to `todoruleid`
-
+As you can see, semgrep tests automatically avoid failing on lines marked with `# todoruleid` or `# todook`.
 To avoid failing on TODOs you can specify `--test-ignore-todo`:
 
+To store rules and test targets in different directories you can specify `--config`.
+For the following directory structure,
+
 ```sh
-python -m semgrep --quiet --test --test-ignore-todo rules/
+$ tree tests
+
+tests
+├── rules
+│   └── python
+│       └── insecure-eval-use.yaml
+└── targets
+    └── python
+        └── insecure-eval-use.py
+
+4 directories, 2 files
 ```
 
-This will produce the following output:
+the command
+
 ```sh
-1 yaml files tested
-check id scoring:
---------------------------------------------------------------------------------
-(TODO: 2) rules/detect-eval.yaml
-	✔ insecure-eval-use                                  TP: 1 TN: 1 FP: 0 FN: 0
---------------------------------------------------------------------------------
-final confusion matrix: TP: 1 TN: 1 FP: 0 FN: 0
---------------------------------------------------------------------------------
+semgrep --test --config tests/rules/ tests/targets/
 ```
 
-To store rules and test targets in different directories you can specify `--config`:
+will produce the same output as before.
 
-```sh
-tree tests
+The subdirectory structure of these two directories must be the same for Semgrep to correctly find the associated files.
+
+To test the autofix behaviour, you can add the autofix test file `rules/detect-eval.fixed.py` to represent the expected outcome of applying the fix to the test code:
```

Comment:
```suggestion
To test the autofix behavior, add the autofix test file `rules/detect-eval.fixed.py` to represent the expected outcome of applying the fix to the test code:
```
Removed "can" because the action is a "must" :)